staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3075,6 +3075,50 @@ type ContainerStatus struct {
 	// +featureGate=SupplementalGroupsPolicy
 	// +optional
 	User *ContainerUser `json:"user,omitempty" protobuf:"bytes,13,opt,name=user,casttype=ContainerUser"`
+	// AllocatedResourcesStatus represents the status of various resources
+	// allocated for this Pod.
+	// +featureGate=ResourceHealthStatus
+	// +optional
+	AllocatedResourcesStatus ResourcesStatus `json:"allocatedResourcesStatus,omitempty" protobuf:"bytes,14,rep,name=allocatedResourcesStatus,casttype=ResourcesStatus,castkey=ResourceName"`
+}
+
+type ResourcesStatus map[ResourceName]ResourceStatus
```

Comment:
A slice of  `struct{Name: ResourceName, ResourceStatus:ResourceStatus}` tagged with `listType=map, listMapKey=name` merges per https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps?

Reply:
fixed this one. Rebased to get feedback. Working on another map now